streams/src/test/java/org/apache/kafka/streams/state/internals/MergedSortedCacheKeyValueStoreIteratorTest.java

```diff
@@ -54,4 +65,85 @@ public void shouldIterateOverRange() throws Exception {
         }
     }
 
+
+    @Test
+    public void shouldSkipLargerDeletedCacheValue() throws Exception {
```

Comment:
nit: `Larger` and `Lower` is a bit confusing here, how about `Larger` and `Smaller`?